src/BuiltInTools/AspNetCoreDeltaApplier/StartupHook.cs

```diff
@@ -0,0 +1,142 @@
+            foreach (var deltaReceiverAttribute in deltaReceiverAttributes)
+            {
+                Log($"Attempting to locate receiver {deltaReceiverAttribute.Value} in assembly {assembly}");
+                var type = assembly.GetType(deltaReceiverAttribute.Value!, throwOnError: false);
+                if (type is null)
+                {
+                    Log($"Could not find delta receiver type {deltaReceiverAttribute.Value} in assembly {assembly}.");
+                    continue;
+                }
+
+                if (type.GetMethod("DeltaApplied", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static) is MethodInfo methodInfo)
+                {
+                    var action = methodInfo.CreateDelegate<Action>();
+                    Action safeAction = () =>
+                    {
+                        try 
+                        { 
+                            action(); 
+                        } 
+                        catch (Exception ex) 
+                        { 
+                            Log(ex.ToString()); 
+                        }
+                    };
+
+                    receipients.Add(safeAction);
+                }
+                else
+                {
+                    Log($"Could not find method 'DeltaApplied' on type {type}.");
+                }
+            }
+        }
+
+        return receipients;
+    }
+
+    public static async Task ReceiveDeltas()
+    {
+        Log("Attempting to receive deltas.");
+
+        using var pipeClient = new NamedPipeClientStream(".", "netcore-hot-reload", PipeDirection.InOut, PipeOptions.CurrentUserOnly | PipeOptions.Asynchronous);
+        try
+        {
+            await pipeClient.ConnectAsync(5000);
+            Log("Connected.");
+        }
+        catch (TimeoutException)
+        {
+            Log("Unable to connect to hot-reload server.");
+        }
+
+        List<Action>? receiveDeltaNotifications = null;
+
+        while (pipeClient.IsConnected)
+        {
+            var bytes = new byte[4096];
+            var numBytes = await pipeClient.ReadAsync(bytes);
+
+            var update = JsonSerializer.Deserialize<UpdatePayload>(bytes.AsSpan(0, numBytes));
+            Log("Attempting to apply deltas.");
+
+            try
+            {
+                foreach (var item in update.Deltas)
+                {
+                    var assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.Modules.FirstOrDefault() is Module m && m.ModuleVersionId == item.ModuleId);
+                    if (assembly is not null)
+                    {
+                        System.Reflection.Metadata.AssemblyExtensions.ApplyUpdate(assembly, item.MetadataDelta, item.ILDelta, ReadOnlySpan<byte>.Empty);
+                    }
+                }
+
+                // We want to base this off of mvids, but we'll figure that out eventually.
```

Comment:
As in eventually we'll be doing something like `update.ChangedMvid`?

Reply:
Yup. Right now it's an empty collection, but it's a hypothetical idea I'd been playing around with to do different things based on what function gets updated.